extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -273,14 +338,51 @@ oppia.directive('imageWithRegionsEditor', [
           if ($scope.hoveredRegion === null) {
             $scope.hoveredRegion = index;
           }
+          $scope.outOfRegion = false;
+        };
+        $scope.onMouseMoveRegion = function() {
+          if ($scope.userIsCurrentlyDragging ||
+              $scope.userIsCurrentlyResizing) {
+            return;
+          }
+          region = reg = cornerAndDimensionsFromRegionArea(
+            $scope.$parent.value.labeledRegions[
+              $scope.hoveredRegion].region.area);
+          if (!$scope.xDirectionToggled && !$scope.yDirectionToggled) {
+            if ($scope.mouseY <= region.y + $scope.resizeRegionBorder) {
+              $scope.yDirection = 1;
+            } else if ($scope.mouseY >= region.height + region.y -
+                $scope.resizeRegionBorder) {
+              $scope.yDirection = -1;
+            } else {
+              $scope.yDirection = 0;
+            }
+            if ($scope.mouseX <= region.x + $scope.resizeRegionBorder) {
+              $scope.xDirection = 1;
+            } else if ($scope.mouseX >= region.width + region.x -
+                $scope.resizeRegionBorder) {
+              $scope.xDirection = -1;
+            } else {
+              $scope.xDirection = 0;
+            }
+          }
         };
         $scope.onMouseoutRegion = function(index) {
           if ($scope.hoveredRegion === index) {
             $scope.hoveredRegion = null;
           }
+          if (!$scope.userIsCurrentlyResizing) {
+            $scope.xDirection = 0;
+            $scope.yDirection = 0;
+          }
+          $scope.outOfRegion = true;
         };
         $scope.onMousedownRegion = function() {
           $scope.userIsCurrentlyDragging = true;
+          if ($scope.xDirection || $scope.yDirection) {
+            $scope.userIsCurrentlyDragging = false;
+            $scope.userIsCurrentlyResizing = true;
+          }
```

Comment:
A bit awkward -- $scope.userIsCurrentlyDragging gets flipped from true to false immediately. Can you just make an if/else condition? I don't think you need to be manually setting anything to false here